drivers/otp/Kconfig

```
@@ -0,0 +1,37 @@
+# OTP driver configuration options
+
+# Copyright (c) 2025 STMicroelectronics
+# SPDX-License-Identifier: Apache-2.0
+
+menuconfig OTP
+	bool "One Time Programmable (OTP) handling drivers [EXPERIMENTAL]"
```

Nit: (handling seems kinda redundant for a driver)
```suggestion
	bool "One Time Programmable (OTP) drivers [EXPERIMENTAL]"
```